docs/csharp/whats-new/tutorials/static-abstract-interface-methods.md

```diff
@@ -0,0 +1,147 @@
+
+```csharp
+public record Point<T>(T X, T Y) where T : IAdditionOperators<T, T, T>
+```
+
+After adding that constraint, your `Point<T>` class can use the `+` for its addition operator. Add the same constraint on the `Translation<T>` declaration:
+
+```csharp
+public record Translation<T>(T XOffset, T YOffset) where T : IAdditionOperators<T, T, T>;
+```
+
+The `IAdditionOperators<T,T,T>` constraint prevents a developer using your class from creating a `Translation` using a type that doesn't meet the constraint for the addition to a point. You've added the necessary constraints to the type parameter for `Translation<T>` and `Point<T>` so this code works. You can test by adding code like the following above the declarations of `Translation` and `Point` in your *Program.cs* file:
+
+:::code language="csharp" source="./snippets/staticinterfaces/Program.cs" id="TestAddition":::
+
+You can make this code more reusable by declaring that these types implement the appropriate arithmetic interfaces. The first change to make is to declare that `Point<T, T>` implements the `IAdditionOperators<Point<T>, Translation<T>, Point<T>>` interface. The `Point` type makes use of different types for operands and the result. The `Point` type already implements an `operator +` with that signature, so adding the interface to the declaration is all you need:
+
+```csharp
+public record Point<T>(T X, T Y) : IAdditionOperators<Point<T>, Translation<T>, Point<T>>
+    where T : IAdditionOperators<T, T, T>
+```
+
+Finally, when you're performing addition, it's useful to have a property that defines the additive identity value for that type. There's a new experimental interface for that feature: `IAdditiveIdentity<TSelf, TResult>`. A translation of `{0, 0}` is the additive identity: The resulting point is the same as the left operand. The `IAdditiveIdentity<TSelf, TResult>` interface defines one readonly property: `AdditiveIdentity`, that returns the identity value. The `Translation<T>` needs a few changes to implement this interface:
+
+:::code language="csharp" source="./snippets/staticinterfaces/Translation.cs":::
+
+There are a few changes here, so let's walk through them one by one. First, you declare that the `Translation` type implements the `IAdditiveIdentity` interface:
+
+```csharp
+public record Translation<T>(T XOffset, T YOffset) : IAdditiveIdentity<Translation<T>, Translation<T>> 
+```
+
+You next might try implementing the interface member as shown in the following code:
+
+```csharp
+public static Translation<T> AdditiveIdentity =>
+    new Translation<T>(XOffset: 0, YOffset: 0);
+```
+
+The preceding code won't compile, because `0` depends on the type. The answer: Use `IAdditiveIdentity<T>.AdditiveIdentity` for `0`. That change means that your constraints must now include that `T` implements `IAdditiveIdentity<T>`. Now that you've added that constraint on `Translation<T>`, you need to add the same constraint to `Point<T>`:
```

Comment:
> The answer: Use `IAdditiveIdentity<T>.AdditiveIdentity` for `0`.

Please insert a code-snippet showing this, to make it very very clear (it's obvious what's meant, but anyway to be complete).